### PR TITLE
Hardening/0136 safe bson operations 2

### PR DESCRIPTION
**Pull request by fgalan**
Finished #136 implementation

Appart from that, unit test broken in a previous PR today has been fixed.

### CODE REVIEW

File: test/unittests/testInit.h [hunk 1]
```diff
@@ -43,6 +45,8 @@
 #define RES_CER_STATUS(i)       res.contextElementResponseVector.get(i)->statusCode
 #define RES_CER_ATTR(i, j)      res.contextElementResponseVector.get(i)->contextElement.contextAttributeVector.get(j)
 
+#define C_STR_FIELD(b, f)       getStringField(b, f).c_str();
```
Comment (fgalan) [inline test/unittests/testInit.h]: Although removed from the non-test code base, the macro still have sense in the unit test code.

NTC

File: test/unittests/testInit.h [hunk 2]
```diff
@@ -43,6 +45,8 @@
 #define RES_CER_STATUS(i)       res.contextElementResponseVector.get(i)->statusCode
 #define RES_CER_ATTR(i, j)      res.contextElementResponseVector.get(i)->contextElement.contextAttributeVector.get(j)
 
+#define C_STR_FIELD(b, f)       getStringField(b, f).c_str()
```
Comment (fgalan) [inline test/unittests/testInit.h]: Although removed from the non-test code base, the macro still have sense in the unit test code.

NTC

File: src/lib/serviceRoutines/postQueryContext.cpp [hunk 3]
```diff
@@ -155,7 +155,7 @@ static void queryForward(ConnectionInfo* ciP, QueryContextRequest* qcrP, Format
   if ((out == "error") || (out == ""))
   {
     qcrsP->errorCode.fill(SccContextElementNotFound, "");
-    LM_E(("Runtime Error (error forwarding 'Query' to providing application)"));
+    LM_W(("Runtime Error (error forwarding 'Query' to providing application)"));
```
Comment (fgalan) [inline src/lib/serviceRoutines/postQueryContext.cpp]: This is not actually part of the topic of the PR, but I have seem while fixing other things.

This condition shouldn't be loged as ERROR, as it is not a problem in the CB itself, but in a external system. Thus, I think that the righ log level is WARNING.

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**